### DISCUSSION TITLE
Can you automatically map relative path from the destination?

### DISCUSSION THREAD

**Discussion by johanneszellinger** [dotnet/yarp#1680] [Q&A] [answered]
Please forgive my ignorance, first time I'm in need of a proxy and still have to wrap my head around a few things.

My question is the following:
My site (blazor server site, using YARP via nuget) is running at https://myawesomesite.test
I want to inject a webserver running at http://localhost:8080 any time the navigates to https://myawesomesite.test/myproxy

I have the following config in appsettings.json

```
  "ReverseProxy": {
    "Routes": {
      "proxy-test": {
        "ClusterId": "proxytest-cluster",
        "Match": {
          "Path": "/myproxy/{**catch-all}"
        },
        "Transforms": [
          { "PathRemovePrefix": "/myproxy" }
        ]
      }
    },
    "Clusters": {
      "proxytest-cluster": {
        "Destinations": {
          "proxytest-destination": {
            "Address": "http://localhost:8080"
          }
        }
      }
    }
  }
```
So far so good, however the local webserver has some static files, such as images, css and json:

```
http://localhost:8080
  /img
    /img01.jpg
    /img02.jpg
    /...
  /js
    /somefunctions.js
    /someotherfunctions.js
  /css
    /design.css
```
These are not mapped correctly, ending up in a 404 response, since for example the path https://myawesomesite.test/img/img01.jpg does not exist. 

I tested around adding extra routes for these folders, e.g.

```
      "return": {
        "ClusterId": "proxytest-cluster",
        "Match": {
          "Path": "/img/{**catch-all}"
        }
      }
```
Which seems to work, however

1. This seems tedious to do, especially since the actual service I want to proxy to is under active development (not by me) and some path might change
2. some "return" routes might interfere with other content on my own site. 

So the question is, how this is normally done? Is it possible to automate the mapping of these paths in the response html?

**Comment by Tratcher** [comment, marked as answer]
Check out https://github.com/microsoft/reverse-proxy/discussions/717

**Reply by johanneszellinger** [reply]
Hi, thanks. I already had a look at this thread. So if I understood correctly, it basically says that in case the proxied website uses static routs, there isn't much one can do, right?
My external service is CVAT running in a local docker container, which actually has a section in the [docs on how to set it up behind a proxy](https://openvinotoolkit.github.io/cvat/docs/administration/basics/installation/#deploying-cvat-behind-a-proxy).
I'm guessing, this might be what I have to look into?

**Reply by Tratcher** [reply]
Those proxy settings sound like they're for outbound traffic, not inbound. You're right though that this is best handled in the destination app when generating the links.
